softlearning/environments/adapters/dm_control_adapter.py

```diff
@@ -0,0 +1,171 @@
+        if env is None:
+            assert (domain is not None and task is not None), (domain, task)
+            env = suite.load(
+                domain_name=domain,
+                task_name=task,
+                task_kwargs=kwargs
+                # TODO(hartikainen): Figure out how to pass kwargs to this guy.
+                # Need to split into `task_kwargs`, `environment_kwargs`, and `visualize_reward` bool.
+                # Check the suite.load(.) in:
+                # https://github.com/deepmind/dm_control/blob/master/dm_control/suite/__init__.py
+            )
+        else:
+            assert domain is None and task is None, (domain, task)
+
+        assert isinstance(env.observation_spec(), OrderedDict)
+        self.observation_keys = (
+            observation_keys or tuple(env.observation_spec().keys()))
+
+        # Ensure action space is already normalized.
+        if normalize:
+            np.testing.assert_equal(env.action_spec().minimum, -1)
+            np.testing.assert_equal(env.action_spec().maximum, 1)
+
+        self._env = env
+
+    @property
+    def observation_space(self):
+        observation_space = convert_dm_control_to_gym_space(self._env.observation_spec())
+        return observation_space
+
+    @property
+    def active_observation_shape(self):
+        """Shape for the active observation based on observation_keys."""
+        observation_space = self.observation_space
+        active_size = sum(
+            np.prod(observation_space.spaces[key].shape)
+            for key in self.observation_keys)
+        active_shape = (int(active_size), )
+        return active_shape
+
+    def convert_to_active_observation(self, observation):
+        flattened_observation = np.concatenate([
+            observation[key] for key in self.observation_keys], axis=-1)
+        return flattened_observation
+
+    @property
+    def action_space(self, *args, **kwargs):
+        action_space = convert_dm_control_to_gym_space(self._env.action_spec())
+        if len(action_space.shape) > 1:
+            raise NotImplementedError(
+                "Action space ({}) is not flat, make sure to check the"
+                " implemenation.".format(action_space))
+        return action_space
+
+    def step(self, action, *args, **kwargs):
+        timestep = self._env.step(action, *args, **kwargs)
+        observation = timestep.observation
+        reward = timestep.reward
+        terminal = timestep.last()
+        info = {}
+        #TODO(Alacarter): See if there's a way to pull info from the environment.
+        return observation, reward, terminal, info
+
+    def reset(self, *args, **kwargs):
+        timestep = self._env.reset(*args, **kwargs)
+        return timestep.observation
+
+    def render(self, *args, mode="human", camera_id=0, **kwargs):
+        if mode == "human":
+            #TODO(Alacarter): Figure out how to not continuously launch viewers if one is already open.
```

Comment:
You could have this todo inside the `NotImplementedError` to make it a bit clearer what's happening:
```
raise NotImplementedError(
    "TODO(Alacarter): Figure out how to not continuously launch"
    " viewers if one is already open. <also add link to github issue here>.")
```